src/dir.cr

```diff
@@ -279,6 +279,14 @@ class Dir
     Crystal::System::Dir.delete(path.to_s)
   end
 
+  # Removes the directory at the given path.
+  def self.delete?(path : Path | String) : Bool
+    delete(path)
+    true
+  rescue File::NotFoundError
```

Comment:
The same can be done here too to avoid raising in the first place.

Also, another option instead of a `raise_on_missing` argument is to do something like:
```cr
def delete(path) : Nil
  delete?(path) || raise "..."
end
```
For me it is more idiomatic, this kind of logic is already present in the stdlib.

Reply:
Ok,but then I need to duplicate the exception string *exactly* or the tests fail.

Reply:
I meant that it will be done in `Crystal::System::Dir.delete?(path) : Bool`, which won't raise any exception.
There is only one error possible so it can be done.

Reply:
This can't be solved with a `System::Dir::delete?` returning `Bool`.  `delete?` returns `false` for `ENOENT` and raises for all other errors.

I suppose `System...delete?` could return `Errno`?

Reply:
I believe it is fine to make it return `Bool` – `true` when the directory/file is present `false`, otherwise – and keep it exceptionnaly raise for other, less likely errors.

Reply:
That's also how you implemented `delete?` in `Dir` and `File` at the top-level at first: only rescue `File::NotFoundError` exceptions.

Reply:
I like what this PR brings and what @j8r said: it makes sense to port the same approach to `system/*/dir.cr` and code something like:

```crystal
def self.delete(path, *, raise_on_missing : Bool) : Bool
    err = LibC.rmdir(path.check_no_null_byte)

    if err != -1
      true
    elsif !raise_on_missing && Errno.value == Errno::ENOENT
      false
    else
      raise ::File::Error.from_errno("Error deleting dir", file: path)
    end
end
```

⚠️ I'm not paying attention to details!

What do you think @didactic-drunk ?

Reply:
Sure.  @beta-ziliani, you're new to me.  This will be approved or still needs @asterite approval?

Reply:
Nice to meet you! You'll find me in the [team section](https://www.crystal-lang.org/team) ;-)
Any PR needs *two* approvals from core team members (see the aforementioned link). I think this is close to get the approvals. Personally, I'd like to see the change proposed above, but I can take care of that if you prefer.

Reply:
Is it preferable to have an `raise_on_missing` argument or have two methods instead, one that can return nil and the other calling it and raising on nil. A bit like `#to_i?` and `#to_i`, or `Enumerable#first?` and `Enumerable#first` (raising).

Reply:
A branch that potentially raises, even if that's impossible, is technically worse that code that can never raise.

Reply:
From an API standpoint, it doesn't matter as this is about the internals. From a technical point, it's an optimization that could be done later. It's already making a difference IMO.

Reply:
Right. I think what's important in this PR is to have something working. The performance differences or styles will probably not matter unless you are deleting files in a hot loop, which doesn't sound like something usual.

Reply:
@j8r I agree 1 method without an argument is a better approach, however the specs require a specific error message to pass.  Either raise is duplicated exactly and kept in sync should the message ever change or an argument is used.  Or the specs are changed.  Or errors are reworked somehow.

My approach was I think the least invasive and I hope fastest to get approved.  Past experience has shown my more complicated PR's stagnate for **very** long periods.

My preference is:
1) Get this merged so I can begin using it
2) Let someone else optimize it later and take the burden of keeping error messages in sync or coming up with another solution